File: cpp/questions/twoSum.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> hm; 
        for (int i=0; i<nums.size(); ++i) {
            const int n = nums[i];
            auto it = hm.find(target-n);
            if (it != hm.end()) {
                return {i, it->second};
            }
            hm[n] = i;
        }
        return {};
    }
};

File: python/questions/twoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dt = dict()
        for i in range(len(nums)):
            tmn = target - nums[i]
            cp = dt.get(tmn, -1)
            if cp != -1:
                return [i, cp]
            dt[nums[i]] = i
        return []
